packages/client/hmi-client/src/components/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -736,6 +754,43 @@ const applyConfigValues = (config: ModelConfiguration) => {
 	logger.success(`Configuration applied ${config.name}`);
 };
 
+const onEditName = () => {
+	isEditingName.value = true;
+	newName.value = knobs.value.transientModelConfig.name ?? '';
+};
+
+const onEditDescription = () => {
+	isEditingDescription.value = true;
+	newDesciption.value = knobs.value.transientModelConfig.description ?? '';
+};
+
+const onConfirmEditName = () => {
+	knobs.value.transientModelConfig.name = newName.value;
+	isEditingName.value = false;
+};
+
+const onConfirmEditDescription = () => {
+	knobs.value.transientModelConfig.description = newDesciption.value;
+	isEditingDescription.value = false;
+};
```

Comment:
Can you just edit the properties in `knobs.value` instead of creating new ref variables? Also the `?? ''` doesn't look like it's needed since you are assigning them to '' in the beginning

Reply:
Since we can undo the edits I still want the original knobs.value `name` and `description`.  I assign them if its confirmed

Reply:
The type of `ModelConfiguration` has name and description as optional so Im sort of forced to have a fallback
newName: string 
newDescription: string

ModelConfiguration.name: string | undefined
ModelConfiguration.description: string | undefined

Reply:
ah ok looks good then